src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -90,62 +105,119 @@ public ShooterSubsystem(ShooterHoodSubsystem shooterHood) {
     shooterPIDRight.setIZone(kIz, 0);
     shooterPIDLeft.setFF(kFF, 0);
     shooterPIDRight.setFF(kFF, 0);
+
     shooterPIDLeft.setOutputRange(-outputRange, outputRange, 0);
     shooterPIDRight.setOutputRange(-outputRange, outputRange, 0);
     // Sets the shooter motor to coast so that subsequent shots don't have to rev up
     // from 0 speed.
-    shooterMAXLeft.setIdleMode(IdleMode.kCoast);
-    shooterMAXRight.setIdleMode(IdleMode.kCoast);
-    shooterMAXLeft.setSmartCurrentLimit(currentLimit);
-    shooterMAXRight.setSmartCurrentLimit(currentLimit);
+    follower_shooterMAXLeft.setIdleMode(IdleMode.kCoast);
+    leader_shooterMAXRight.setIdleMode(IdleMode.kCoast);
+    follower_shooterMAXLeft.setSmartCurrentLimit(currentLimit);
+    leader_shooterMAXRight.setSmartCurrentLimit(currentLimit);
 
     // The shooter should idle and run in the positive direction for normal
     // operation.
     // Flip this invert setting if it runs backwards.
-    shooterMAXRight.setInverted(false);
+    leader_shooterMAXRight.setInverted(false);
 
     // Sets the left shooter motor to follow the right motor, and be inverted.
-    shooterMAXLeft.follow(shooterMAXRight, true);
+    follower_shooterMAXLeft.follow(leader_shooterMAXRight, true);
+
+    lastPos = shooterEncoder.getPosition();
+    lastTime = Timer.getFPGATimestamp();
   }
 
-  /**
-   * Passes a preset velocity to the SparkMAX PIDF controller and lets it manage
-   * the NEO's velocity. Intended to be called when a button is pressed.
-   */
   public void shoot() {
-    // extend hood
     // fast shooter wheel
     // run gate if both of "" are good
+    // NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(3);
+
+    if (enablePID) {
+      shooterPIDRight.setReference(MoPrefs.getShooterPIDSetpoint(), ControlType.kVelocity);
```

Comment:
Opportunity to coalesce PID and non-PID setpoints.  The ranges are mappable so there should only be one constant to control both.  Otherwise a maintenance problem.